core/access_tokens.go

```diff
@@ -87,6 +88,15 @@ func (a *API) deleteAccessToken(ctx context.Context, x struct{ ID string }) erro
 	if currentID == x.ID {
 		return errCurrentToken
 	}
-	return a.accessTokens.Delete(ctx, x.ID)
-	// TODO(tessr): delete any associated grants, as well
+	err := a.accessTokens.Delete(ctx, x.ID)
+	if err != nil {
+		return err
+	}
+
+	err = a.revokeGrantsByAccessToken(ctx, x.ID)
+	if err != nil {
+		// well, technically we did delete the access token, so don't return the error
+		log.Printkv(ctx, log.KeyError, "revoking grants for access token %s", x.ID)
+	}
```

Comment:
Oh but in the meantime, I am not sure whether this should return an error. We don't have a way to indicate "hey your thing worked, but you should retry it anyway".

I would lean toward not returning an error for the reasons @tessr mentioned. The primary operation actually did complete.